src/types/register.ts

```diff
@@ -21,4 +23,8 @@ export type RegisteredRejectionType = Register extends { rejections: infer TReje
 
 export type RegisteredRouterState = Register extends { state: infer TState }
   ? TState
-  : {}
\ No newline at end of file
+  : {}
+
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-arguments
```

Comment:
```suggestion
// Because RegisteredRoutes defaults to `[]` it thinks passing it is unnecessary
// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-arguments
```